invoke-ai/InvokeAI
# [feat] Bake in sdxl-vae-fp16-fix model on checkpoint conversion

## What type of PR is this? (check all applicable)


## Have you discussed this change with the InvokeAI team?
      
## Have you updated all relevant documentation?
- [X] No -- no obvious place to put this

## Description

To avoid  the ser having to remember to switch to VAE `fp32` whenever rendering with SDXL models, this PR automatically replaces the default fp32-only VAE with `sdxl-vae-fp16-fix` during the checkpoint conversion step. This will enable all SDXL `.safetensors` files to be run in `fp16` VAE mode. However, it does not affect diffusers models downloaded from HugginFace. These will have to use a pinned VAE in `models.yaml`.

## Related Tickets & Documents




- Related Issue #4454
- Closes #

## QA Instructions, Screenshots, Recordings

1. Remove the `models/.cache` directory so that safetensors files will be re-converted.
2. Make sure that `sdxl-vae-fp16-fix` is installed in `models/sdxl/vae`
3. Load an SDXL `.safetensors` model
4. Test `fp16` when the "Default" VAE is used.



## Added/updated tests?

- [X] No : performed functional testing, but hard to test in an automated fashion without a full model to convert

## [optional] Are there any post deployment tasks we need to perform?
